service/src/main/java/org/kiwiproject/champagne/dao/BuildDao.java

```diff
@@ -4,72 +4,48 @@
 import static org.kiwiproject.base.KiwiStrings.f;
 import static org.kiwiproject.champagne.dao.DaoHelper.LIKE_QUERY_FORMAT;
 
+import java.util.List;
+
 import org.jdbi.v3.sqlobject.customizer.Bind;
 import org.jdbi.v3.sqlobject.customizer.BindBean;
 import org.jdbi.v3.sqlobject.statement.GetGeneratedKeys;
 import org.jdbi.v3.sqlobject.statement.SqlQuery;
 import org.jdbi.v3.sqlobject.statement.SqlUpdate;
 import org.kiwiproject.champagne.model.Build;
 
-import java.util.List;
-
 public interface BuildDao {
     
-    default List<Build> findPagedBuilds(int offset, int limit, long systemId, String componentIdentifierFilter, String componentVersionFilter) {
-        if (isBlank(componentIdentifierFilter) && isBlank(componentVersionFilter)) {
+    default List<Build> findPagedBuilds(int offset, int limit, long systemId, String componentFilter) {
+        if (isBlank(componentFilter)) {
             // No filtering
             return findPagedBuildsNoFilter(offset, limit, systemId);
-        } else if (isBlank(componentIdentifierFilter)) {
-            // Version only filter
-            return findPagedBuildsWithVersionFilter(offset, limit, systemId, f(LIKE_QUERY_FORMAT, componentVersionFilter));
-        } else if (isBlank(componentVersionFilter)) {
-            // Name only filter
-            return findPagedBuildsWithIdentiferFilter(offset, limit, systemId, f(LIKE_QUERY_FORMAT, componentIdentifierFilter));
         }
 
         // Both name and version filtering
-        return findPagedBuildsWithVersionAndIdentifierFilters(offset, limit, systemId, componentIdentifierFilter, componentVersionFilter);
+        return findPagedBuildsWithVersionOrIdentifierFilters(offset, limit, systemId, f(LIKE_QUERY_FORMAT, componentFilter));
```

Comment:
So, could there be a SQL injection hiding here?

Reply:
It shouldn't be as we are still using JDBI's argument replacer